Serializer/JMSSerializerAdapter.php

```diff
@@ -104,7 +104,7 @@ private function convertContext(Context $context, $direction)
         if (null !== $context->getGroups()) {
             $jmsContext->setGroups($context->getGroups());
         }
-        if (null !== $context->getMaxDepth(false) || null !== $context->isMaxDepthEnabled()) {
+        if (null !== $context->getMaxDepth(false) || (null !== $context->isMaxDepthEnabled() && true === $context->isMaxDepthEnabled())) {
```

Comment:
As you are doing a strict type check, I believe the null check can be removed?

Reply:
You're right, I've removed it.